src/darwin/Framework/CHIP/MTRSetupPayload.h

```diff
@@ -49,36 +49,76 @@ typedef NS_ENUM(NSUInteger, MTROptionalQRCodeInfoType) {
 };
 
 /**
- * An optional information item present in the QR code the setup payload was
- * initialized from.
+ * An optional information item present in the setup payload.
+ *
+ * Note that while the Matter specification allows elements containing
+ * arbitrary TLV data types, this implementation currently only supports
+ * String and Int32 values.
+ *
+ * Objects of this type are immutable; calling any deprecated property
+ * setters has no effect.
  */
 MTR_AVAILABLE(ios(16.1), macos(13.0), watchos(9.1), tvos(16.1))
-@interface MTROptionalQRCodeInfo : NSObject
+@interface MTROptionalQRCodeInfo : NSObject <NSCopying>
+
+- (instancetype)initWithTag:(uint8_t)tag stringValue:(NSString *)value MTR_NEWLY_AVAILABLE;
+- (instancetype)initWithTag:(uint8_t)tag int32Value:(int32_t)value MTR_NEWLY_AVAILABLE;
```

Comment:
I think we’re sort of stuck with that convention at least for things like NodeID, but I thing for other values we should start moving away from this. It’s just not idiomatic and bridges badly to Swift. But we can discuss offline.

Reply:
Let's continue to use NSNumber *, as we are an Objective C framework. If we want swift friendly interfaces, we should use refined for swift, not mix swift and objective C paradigms in the same framework, and Objective C frameworks use NSNumber *, and have since our 64 bit transition.